src/progress/progress_set.rs

```diff
@@ -77,17 +81,17 @@ where
     }
 }
 
-impl From<ConfState> for Configuration {
-    fn from(conf_state: ConfState) -> Self {
+impl From<&ConfState> for Configuration {
+    fn from(conf_state: &ConfState) -> Self {
```

Comment:
It's generally not idiomatic to implicitly clone, better to use the `ConfState` and let the caller clone.

Reply:
I guess `cloned` won't cause real clone because `u64` is `Copy`. But I would like to use `.map(|n| *n)` to replace `.cloned()` to make it more clearly.

Reply:
Clippy suggests `cloned` instead of `map(|n| *n)` so I revert it. However I have checked the implementation of `Iterator::cloned`, which makes me sure about there won't be any `clone` here. So could I keep it?